src/status_im/contexts/communities/events.cljs

```diff
@@ -219,3 +230,121 @@
+  [{:keys [db]} [{:keys [communities]}]]
+  (when-let [community (first communities)]
+    {:db (-> db
+             (assoc-in [:communities (:id community) :spectated] true)
+             (assoc-in [:communities (:id community) :spectating] false))
+     :fx [[:dispatch [:communities/handle-community community]]
+          [:dispatch [::mailserver/request-messages]]]}))
+
+(rf/reg-event-fx :chat.ui/spectate-community-success spectate-community-success)
+
+(defn spectate-community-failed
+  [{:keys [db]} [community-id]]
+  {:db (assoc-in db [:communities community-id :spectating] false)})
+
+(rf/reg-event-fx :chat.ui/spectate-community-failed spectate-community-failed)
+
+(defn spectate-community
+  [{:keys [db]} [community-id]]
+  (let [{:keys [spectated spectating joined]} (get-in db [:communities community-id])]
+    (when (and (not joined) (not spectated) (not spectating))
+      {:db            (assoc-in db [:communities community-id :spectating] true)
+       :json-rpc/call [{:method     "wakuext_spectateCommunity"
+                        :params     [community-id]
+                        :on-success [:chat.ui/spectate-community-success]
+                        :on-error   (fn [err]
+                                      (log/error {:message
+                                                  "Failed to spectate community"
+                                                  :error err})
+                                      (rf/dispatch [:chat.ui/spectate-community-failed
+                                                    community-id]))}]})))
+
+(schema/=> spectate-community
+  [:=>
+   [:catn
+    [:cofx :schema.re-frame/cofx]
+    [:args
+     [:schema [:catn [:community-id [:? :string]]]]]]
+   [:maybe
+    [:map
+     [:db map?]
+     [:json-rpc/call :schema.common/rpc-call]]]])
+
+(rf/reg-event-fx :chat.ui/spectate-community spectate-community)
+
+(rf/defn navigate-to-serialized-community
+  [_ {:keys [community-id]}]
+  (native-module/deserialize-and-compress-key
+   community-id
+   (fn [deserialized-key]
+     (rf/dispatch [:communities/navigate-to-community-overview (str deserialized-key)]))))
+
+(rf/reg-event-fx :communities/navigate-to-community-overview
+ (fn [cofx [deserialized-key]]
+   (if (string/starts-with? deserialized-key "z")
+     (navigate-to-serialized-community cofx deserialized-key)
+     (rf/merge
+      cofx
+      {:fx [[:communities/fetch-community deserialized-key]
+            [:dispatch [:navigate-to :community-overview deserialized-key]]]}
+      (navigation/pop-to-root :shell-stack)))))
+
+(rf/reg-event-fx :communities/navigate-to-community-chat
+ (fn [{:keys [db]} [chat-id pop-to-root?]]
+   (let [chat         (get-in db [:chats chat-id])
+         community-id (get chat :community-id)]
+     {:fx [(when community-id
+             [:dispatch [:communities/fetch-community community-id]])
+           (if pop-to-root?
+             [:dispatch [:universal-links/handle-community-chat chat]]
```

Comment:
Why another dispatch? We should directly call `chat/pop-to-root-and-navigate-to-chat` here.
(Also use of universal-links looks confusing).
I think `universal-links` is handler, which uses communities, chats etc. to handle links. Not other way around.